1.segment_images/scripts/2.segment_whole_organoids.py

```diff
@@ -0,0 +1,243 @@
+import sys
+
+import matplotlib.pyplot as plt
+
+# Import dependencies
+import numpy as np
+import skimage
+import tifffile
+from cellpose import core
+from cellpose import io as cellpose_io
+from cellpose import models
+
+cellpose_io.logger_setup()
+import torch
+from cellpose.io import imread
+from PIL import Image
+from skimage import io
+
+use_GPU = torch.cuda.is_available()
+
+# set import path
+sys.path.append(str(pathlib.Path("../../utils/").resolve()))
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+print(in_notebook)
+
+
+# In[2]:
+
+
+if not in_notebook:
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+    parser.add_argument(
+        "--window_size", type=int, help="Size of the window to use for the segmentation"
+    )
+    parser.add_argument(
+        "--clip_limit",
+        type=float,
+        help="Clip limit for the adaptive histogram equalization",
+    )
+
+    args = parser.parse_args()
+    window_size = args.window_size
+    clip_limit = args.clip_limit
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+
+else:
+    input_dir = pathlib.Path("../../data/z-stack_images/C4-2/").resolve(strict=True)
+    window_size = 3
+    clip_limit = 0.1
+
+mask_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
+mask_path.mkdir(exist_ok=True, parents=True)
+
+
+# ## Set up images, paths and functions
+
+# In[3]:
+
+
+image_extensions = {".tif", ".tiff"}
+files = sorted(input_dir.glob("*"))
+files = [str(x) for x in files if x.suffix in image_extensions]
+
+
+# In[4]:
+
+
+# find the cytoplasmic channels in the image set
+for f in files:
+    if "405" in f:
+        nuclei = io.imread(f)
+    elif "488" in f:
+        cyto1 = io.imread(f)
+    elif "555" in f:
+        cyto2 = io.imread(f)
+    elif "640" in f:
+        cyto3 = io.imread(f)
+    elif "TRANS" in f:
+        brightfield = io.imread(f)
+    else:
+        print(f"Unknown channel: {f}")
+
+cyto = np.max([cyto1, cyto2, cyto3], axis=0)
+# pick which channels to use for cellpose
+cyto = skimage.exposure.equalize_adapthist(cyto1, clip_limit=clip_limit)
+
+
+original_cyto_image = cyto.copy()
+
+original_cyto_z_count = cyto.shape[0]
+
+
+# In[5]:
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+image_stack_2_5D = np.empty((0, cyto.shape[1], cyto.shape[2]), dtype=cyto.dtype)
+for image_index in range(cyto.shape[0]):
+    image_stack_window = cyto[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # guassian blur the image stack
+    image_stack_window = skimage.filters.gaussian(image_stack_window, sigma=1)
+    # max project the image stack
```

Comment:
Consider providing  an explanation why do we need to `max project the image stack`